### PR TITLE
Vd photogram authorization

### PR DESCRIPTION
**Pull request by viviancreates**

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -58,13 +59,21 @@ def destroy
...
   private
-    # Use callbacks to share common setup or constraints between actions.
-    def set_comment
-      @comment = Comment.find(params[:id])
-    end
 
-    # Only allow a list of trusted parameters through.
-    def comment_params
-      params.require(:comment).permit(:author_id, :photo_id, :body)
+  # Use callbacks to share common setup or constraints between actions.
```
Comment (tluu5) [inline app/controllers/comments_controller.rb:63]: Great job settling up def set_comment

File: app/controllers/likes_controller.rb [hunk 2]
```diff
@@ -1,5 +1,10 @@
 class LikesController < ApplicationController
   before_action :set_like, only: %i[ show edit update destroy ]
+  before_action : is_an_authorized_user, only: [:destroy, :create]
+
+  def is_an_authorized_user
```
Comment (tluu5) [inline app/controllers/likes_controller.rb:5]: Great job adding before_action : is_an_authorized_user

File: app/models/comment.rb [hunk 3]
```diff
@@ -22,6 +22,7 @@
 class Comment < ApplicationRecord
   belongs_to :author, class_name: "User", counter_cache: true
   belongs_to :photo, counter_cache: true
-
+  has_one :owner, through: :photo
```
Comment (tluu5) [inline app/models/comment.rb:25]: Cool adding has_one for comment.rb

File: config/routes.rb [hunk 4]
```diff
@@ -4,14 +4,14 @@
   devise_for :users
   
   resources :comments
-  resources :follow_requests
-  resources :likes
-  resources :photos
+  resources :follow_requests, except: [:index, :show, :new, :edit]
```
Comment (tluu5) [inline config/routes.rb:7]: Thumbs up for using except and only on routes.rb

**Review by tluu5 — APPROVED** [review]
You are good to go.

File: app/models/like.rb
Comment (dsimp) [inline app/models/like.rb:1]: Good job setting the relationship between  an owners like to a photo

**Review by dsimp — APPROVED** [review]
Clean formatting!

File: app/controllers/comments_controller.rb [hunk 5]
```diff
@@ -58,13 +59,21 @@ def destroy
...
+  def set_comment
+    @comment = Comment.find(params[:id])
+  end
+
+  def is_an_authorized_user
```
Comment (marywebby) [inline app/controllers/comments_controller.rb:68]: good job!

File: app/controllers/follow_requests_controller.rb [hunk 6]
```diff
@@ -58,6 +59,13 @@ def destroy
   end
 
   private
+
+  def ensure_current_user_is_owner
```
Comment (marywebby) [inline app/controllers/follow_requests_controller.rb:63]: i love this implementation of .sender and .recipient!!! great job!!

File: app/controllers/likes_controller.rb [hunk 7]
```diff
@@ -1,5 +1,10 @@
...
+    if !@like.owner.private? || @like.owner == current_user || current_user.leaders.include?(@like.owner) redirect_back(fallback_location: root_url), alert: "Not authorized")
```
Comment (marywebby) [inline app/controllers/likes_controller.rb:6]: 😁😁

**Review by marywebby — APPROVED** [review]
Great job on this!!!

**Pull request merged**